### PR TITLE
NEXUS-20703 - R format - Impossible to upload package from ui

### PR DESCRIPTION
**Pull request by aornatovskyy**
Ticket: https://issues.sonatype.org/browse/NEXUS-20703

### CODE REVIEW

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RPathUtils.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ private static String match(final TokenMatcher.State state, final String name) {
    * Builds a path to an archive for a particular path and filename.
    */
   static String path(final String path, final String filename) {
-    return path + "/" + filename;
+    return  path.endsWith("/") ? path + filename : path + "/" + filename;
```
Comment (sergeyonofreychuck) [inline nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RPathUtils.java]: return StringUtils.stripEnd(path, "/") + filename;   ???

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RPathUtils.java [hunk 2]
```diff
@@ -77,7 +77,7 @@ static String extractFullPath(final Context context) {
   /**
    * Removes slash if path starts with it
    */
-  private static String removeInitialSlashFromPath(final String path) {
+  static String removeInitialSlashFromPath(final String path) {
```
Comment (sergeyonofreychuck) [inline nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RPathUtils.java]: return StringUtils.stripStart(path, "/"); ???

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/upload/RUploadHandler.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                        final Set<UploadDefinitionExtension> uploadDefinitionExtensions)
+  {
+    super(uploadDefinitionExtensions);
+    this.variableResolverAdapter = variableResolverAdapter;
+    this.contentPermissionChecker = contentPermissionChecker;
+  }
+
+  @Override
+  public UploadResponse handle(final Repository repository, final ComponentUpload upload) throws IOException {
+    final AssetUpload assetUpload = upload.getAssetUploads().get(0);
```
Comment (doddi) [inline nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/upload/RUploadHandler.java]: Question: Do we need to check for null on these parameters `checkNotNull`?

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/upload/RUploadHandler.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    final PartPayload payload = assetUpload.getPayload();
+    final Map<String, String> fields = assetUpload.getFields();
+    final String uploadPath = removeInitialSlashFromPath(fields.get(PATH_ID));
+    final String assetPath = path(uploadPath, payload.getName());
+
+    ensurePermitted(repository.getName(), RFormat.NAME, assetPath, Collections.emptyMap());
+
+    try {
+      UnitOfWork.begin(repository.facet(StorageFacet.class).txSupplier());
+      Asset asset = repository.facet(RHostedFacet.class).upload(assetPath, payload);
```
Comment (doddi) [inline nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/upload/RUploadHandler.java]: Question: What happens if redeploy is disabled and an already existing package is uploaded? Does this return a correct `UploadResponse` i.e. unable to upload?

Comment (aornatovskyy) [reply nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/upload/RUploadHandler.java]: here is the message "Repository does not allow updating assets: r-hosted"

File: nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/upload/RUploadHandler.java (hunk 3 above)
Comment (aornatovskyy) [reply nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/upload/RUploadHandler.java]: It is checked on the abstractions ValidatingComponentUpload and UploadManagerImpl
At the same time, this Handler was created similar to other formats.

**Pull request merged**